src/main/scala/com/cognite/spark/datasource/SdkRddV1.scala

```diff
@@ -0,0 +1,41 @@
+package com.cognite.spark.datasource
+
+import java.util.concurrent.Executors
+
+import cats.effect.IO
+import com.cognite.sdk.scala.v1.GenericClient
+import com.softwaremill.sttp.asynchttpclient.cats.AsyncHttpClientCatsBackend
+import org.apache.spark.{Partition, SparkContext, TaskContext}
+import org.apache.spark.rdd.RDD
+import org.apache.spark.sql.Row
+
+import scala.concurrent.ExecutionContext
+
+case class SdkPartition(cursor: Option[String], size: Option[Int], index: Int) extends Partition
+
+case class SdkRddV1[A](
+    @transient override val sparkContext: SparkContext,
+    config: RelationConfig,
+    rddCompute: (GenericClient[IO, Nothing], SdkPartition) => Iterator[Row],
+    cursors: Iterator[(Option[String], Option[Int])])
+    extends RDD[Row](sparkContext, Nil) {
+  override def getPartitions: Array[Partition] =
+    cursors.toIndexedSeq.zipWithIndex.map {
+      case ((cursor, size), index) =>
+        val partitionSize = (config.limit, size) match {
+          case (None, s) => s
+          case (Some(l), Some(s)) => Some(scala.math.min(l, s))
+          case (l, None) => l
+        }
+        SdkPartition(cursor, partitionSize, index)
+    }.toArray
+  override def compute(_split: Partition, context: TaskContext): Iterator[Row] = {
+    val split = _split.asInstanceOf[SdkPartition]
+    implicit val cs =
+      IO.contextShift(ExecutionContext.fromExecutor(Executors.newCachedThreadPool()))
+    implicit val sttpBackend = AsyncHttpClientCatsBackend[cats.effect.IO]()
+    implicit val auth = config.auth
+    val client = new GenericClient[IO, Nothing]("cdp-spark-datasource")
```

Comment:
Does creating a new client cause a HTTP client pool to be made? We should share the client if that's the case.

Reply:
Shouldn't sharing connection pools be handled in the SDK?

Reply:
Yes, so I assume each Client got a connection pool, and then you share the client across the code. Though I think it's the sttpBackend which stores the connection pool, and at the moment we create a new sttpBackend per partition, which is ineffective, since Spark will probably run multiple partitions per worker (multiple CPU cores). Instead we should share the sttpBackend and maybe the scala sdk client (depends if it actually allocates resources, or if it's just proxies stuff down to the backend).

Reply:
Good points! Made this work with a single sttpBackend and client.